src/GeneratedInjectorDelegator.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-di for the canonical source repository
+ * @copyright Copyright (c) 2019 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-di/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Di;
+
+use Psr\Container\ContainerInterface;
+
+class GeneratedInjectorDelegator
+{
+    /**
+     * @param string $name
+     */
+    public function __invoke(ContainerInterface $container, $name, callable $callback) : InjectorInterface
+    {
+        $config = $container->has('config') ? $container->get('config') : [];
+        $aotConfig = $config['dependencies']['auto']['aot'] ?? [];
+        $namespace = $aotConfig['namespace'] ?? null;
+
+        $injector = $callback();
+
+        $generatedInjector = $namespace . '\\GeneratedInjector';
```

Comment:
I think we should somehow handle a null or non-string namespace value. by extracting this to a method that ensures typesafety:

```php
private function buildDecoratorClassName(string $namespace): string
{
    return $namespace !== '' ? sprintf('%s\\GeneratedInjector', $namespace) : 'GeneratedInjector';
}
```

and then:

```suggestion
        if ($namespace === null) {
            return $injector;
        }
        
        $generatedInjector = $this->buildDecoratorClassName($namespace);
```

Reply:
No needed, if namespace is empty then `GeneratedInjector` is in the global namespace so `\GeneratedInjector` is perfectly fine.